ChallengesWithTestsMarkVII/ChallengesSet04.cs

```diff
@@ -6,47 +6,108 @@ public class ChallengesSet04
     {
         public int AddEvenSubtractOdd(int[] numbers)
         {
-            throw new NotImplementedException();
+            if (numbers == null || numbers.Length == 0)
+             {
+                return 0;
+             }
+            int sum = 0;
+            foreach (int num in numbers)
+            {
+                if (num%2 == 0)
```

Comment:
Just for fun, could you turn this if-statement that takes up 8 lines into a single line?